File: comguanmuonehours/src/main/java/com/guanmu/onehours/activity/LoginActivity.java
package com.guanmu.onehours.activity;

import android.content.Intent;
import android.os.Bundle;
import androidx.fragment.app.FragmentManager;

import com.guanmu.onehours.R;
import com.guanmu.onehours.fragment.FragmentLoginInput;
import com.guanmu.onehours.fragment.FragmentSignUp;
import com.guanmu.onehours.fragment.FragmentSuccess;
import com.guanmu.onehours.fragment.FragmentVerifyCode;


/**
 * A login screen that offers login via email/password.
 */
public class LoginActivity extends BaseActivity implements
        FragmentLoginInput.OnFragmentInteractionListener,
        FragmentSignUp.OnFragmentInteractionListener,
        FragmentVerifyCode.OnFragmentInteractionListener,
        FragmentSuccess.OnFragmentInteractionListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        FragmentLoginInput mFragInput = new FragmentLoginInput();
        getSupportFragmentManager().beginTransaction().add(R.id.fragment_frame, mFragInput).commit();
    }

    @Override
    public void onFragmentSuccess() {
        FragmentSuccess mFragSuccess = new FragmentSuccess();
        getSupportFragmentManager().popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_frame, mFragSuccess).commit();
    }

    @Override
    public void onFragmentAttemptSignUp(Bundle args) {
        FragmentSignUp mFragSignUp = new FragmentSignUp();
        mFragSignUp.setArguments(args);
        getSupportFragmentManager().popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_frame, mFragSignUp).commit();
    }

    @Override
    public void onFragmentRequestVerify() {
        FragmentVerifyCode mFragVerify = new FragmentVerifyCode();
        getSupportFragmentManager().popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_frame, mFragVerify).commit();
    }

    @Override
    public void onFragmentFinish() {
        finish();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
//        SNS.onActivityResult(requestCode, resultCode, data, SNSType.AVOSCloudSNSQQ);
    }
}

File: comguanmuonehours/src/main/java/com/guanmu/onehours/activity/LoginActivity.kt
package com.guanmu.onehours.activity

import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.FragmentManager

import com.guanmu.onehours.R
import com.guanmu.onehours.fragment.FragmentLoginInput
import com.guanmu.onehours.fragment.FragmentSignUp
import com.guanmu.onehours.fragment.FragmentSuccess
import com.guanmu.onehours.fragment.FragmentVerifyCode


/**
 * A login screen that offers login via email/password.
 */
class LoginActivity : BaseActivity(), FragmentLoginInput.OnFragmentInteractionListener, FragmentSignUp.OnFragmentInteractionListener, FragmentVerifyCode.OnFragmentInteractionListener, FragmentSuccess.OnFragmentInteractionListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val mFragInput = FragmentLoginInput()
        supportFragmentManager.beginTransaction().add(R.id.fragment_frame, mFragInput).commit()
    }

    override fun onFragmentSuccess() {
        val mFragSuccess = FragmentSuccess()
        supportFragmentManager.popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)
        supportFragmentManager.beginTransaction().replace(R.id.fragment_frame, mFragSuccess).commit()
    }

    override fun onFragmentAttemptSignUp(args: Bundle) {
        val mFragSignUp = FragmentSignUp()
        mFragSignUp.arguments = args
        supportFragmentManager.popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)
        supportFragmentManager.beginTransaction().replace(R.id.fragment_frame, mFragSignUp).commit()
    }

    override fun onFragmentRequestVerify() {
        val mFragVerify = FragmentVerifyCode()
        supportFragmentManager.popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)
        supportFragmentManager.beginTransaction().replace(R.id.fragment_frame, mFragVerify).commit()
    }

    override fun onFragmentFinish() {
        finish()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        //        SNS.onActivityResult(requestCode, resultCode, data, SNSType.AVOSCloudSNSQQ);
    }
}

